src/core/bpf-firewall.c

```diff
@@ -542,11 +542,73 @@ int bpf_firewall_compile(Unit *u) {
         return 0;
 }
 
+static int load_bpf_progs_from_fs_to_set(Unit *u, char **filter_paths, Set **set) {
+        char **bpf_fs_path;
+        _cleanup_free_ BPFProgram *prog = NULL;
+        int r;
+
+        *set = set_free_with_destructor(*set, bpf_program_unref);
```

Comment:
for new code, please use a "hashops" structure when doing set_new() that has a desctructor call set. With that you can then just do set_free() and the destructor is implicitly called. In fact, with that you can even do set_clear() here, which is a bit nicer I think.

Reply:
Yes, that's better :)